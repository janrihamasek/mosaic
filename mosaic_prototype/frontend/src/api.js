import apiClient from './apiClient';
import { API_BASE_URL as API_BASE } from './config';

function extractParams(filters = {}) {
  const params = new URLSearchParams();
  if (filters.startDate) params.set('start_date', filters.startDate);
  if (filters.endDate) params.set('end_date', filters.endDate);
  if (filters.activity && filters.activity !== 'all') params.set('activity', filters.activity);
  if (filters.category && filters.category !== 'all') params.set('category', filters.category);
  return params;
}

// --- ENTRIES ---
export async function fetchEntries(filters = {}) {
  const params = Object.fromEntries(extractParams(filters));
  const response = await apiClient.get('/entries', { params });
  return response.data;
}

export async function addEntry(entry) {
  const response = await apiClient.post('/add_entry', entry);
  return response.data;
}

export async function deleteEntry(id) {
  const response = await apiClient.delete(`/entries/${id}`);
  return response.data;
}

// --- ACTIVITIES ---
export async function fetchActivities({ all = false } = {}) {
  const response = await apiClient.get('/activities', { params: all ? { all: 'true' } : {} });
  return response.data;
}

export async function addActivity(activity) {
  const response = await apiClient.post('/add_activity', activity);
  return response.data;
}

export async function updateActivity(id, payload) {
  const response = await apiClient.put(`/activities/${id}`, payload);
  return response.data;
}

export async function deactivateActivity(id) {
  const response = await apiClient.patch(`/activities/${id}/deactivate`);
  return response.data;
}

export async function activateActivity(id) {
  const response = await apiClient.patch(`/activities/${id}/activate`);
  return response.data;
}

export async function deleteActivity(id) {
  const response = await apiClient.delete(`/activities/${id}`);
  return response.data;
}

// --- STATS ---
export async function fetchProgressStats({ date } = {}) {
  const params = {};
  if (date) params.date = date;
  const response = await apiClient.get('/stats/progress', { params });
  return response.data;
}

// --- TODAY ---
export async function fetchToday(dateStr) {
  const params = dateStr ? { date: dateStr } : {};
  const response = await apiClient.get('/today', { params });
  return response.data;
}

export async function finalizeDay(dateStr) {
  const response = await apiClient.post('/finalize_day', { date: dateStr });
  return response.data;
}

// --- IMPORT ---
export async function importEntriesCsv(file) {
  const formData = new FormData();
  formData.append('file', file);
  const response = await apiClient.post('/import_csv', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return response.data;
}

export const getStreamProxyUrl = (url, username, password) =>
  `${API_BASE}/api/stream-proxy?url=${encodeURIComponent(url)}&username=${encodeURIComponent(username)}&password=${encodeURIComponent(password)}`;
