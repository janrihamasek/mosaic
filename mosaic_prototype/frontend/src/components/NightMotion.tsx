import React, {
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from "react";
import { useDispatch, useSelector } from "react-redux";
import { useForm } from "react-hook-form";

import { styles } from "../styles/common";
import { useCompactLayout } from "../utils/useBreakpoints";
import FormWrapper from "./shared/FormWrapper";
import { getStreamProxyUrl } from "../api";
import {
  selectNightMotionState,
  setError,
  setField,
  setStatus,
  startStream,
  stopStream,
  type NightMotionStatus,
} from "../store/nightMotionSlice";
import type { AppDispatch } from "../store";

type NotifyVariant = "success" | "error" | "info";

interface NightMotionProps {
  onNotify?: (message: string, variant?: NotifyVariant) => void;
}

interface NightMotionFormValues {
  username: string;
  password: string;
  streamUrl: string;
}

const statusColors: Record<NightMotionStatus, string> = {
  idle: "#888888",
  connecting: "#d0b000",
  playing: "#2ecc71",
  error: "#e74c3c",
};

const statusLabels: Record<NightMotionStatus, string> = {
  idle: "Idle",
  connecting: "Connecting…",
  playing: "Live",
  error: "Error",
};

const STORAGE_KEY = "nightMotionConfig";

const formFieldStyles = {
  display: "flex",
  flexDirection: "column" as const,
  gap: "0.35rem",
};

const passwordToggleStyle = {
  ...styles.button,
  padding: "0.4rem 0.6rem",
  fontSize: "0.8rem",
  backgroundColor: "#444",
};

const videoWrapperBase = {
  ...styles.card,
  display: "flex",
  flexDirection: "column" as const,
  gap: "1rem",
  minHeight: "20rem",
  flex: "1 1 24rem",
  background: "linear-gradient(145deg, #1f2228 0%, #252830 100%)",
};

export default function NightMotion({ onNotify }: NightMotionProps) {
  const dispatch = useDispatch<AppDispatch>();
  const { username, password, streamUrl, status, error } = useSelector(selectNightMotionState);
  const [showPassword, setShowPassword] = useState(false);
  const [statusVisible, setStatusVisible] = useState(true);
  const { isCompact } = useCompactLayout();
  const videoRef = useRef<HTMLVideoElement | null>(null);
  const hasHydratedConfig = useRef(false);

  const {
    register,
    handleSubmit,
    reset,
    formState: { isSubmitting },
  } = useForm<NightMotionFormValues>({
    defaultValues: { username, password, streamUrl },
  });

  const layoutStyle = useMemo(
    () => ({
      display: "flex",
      flexDirection: isCompact ? "column" : "row",
      alignItems: "stretch",
      gap: isCompact ? "1rem" : "1.5rem",
      width: "100%",
    }),
    [isCompact]
  );

  const videoWrapperStyle = useMemo(
    () => ({
      ...videoWrapperBase,
      minHeight: isCompact ? "18rem" : videoWrapperBase.minHeight,
    }),
    [isCompact]
  );

  const videoStyle = useMemo(
    () => ({
      width: "100%",
      height: isCompact ? "16rem" : "18rem",
      backgroundColor: "#000",
      borderRadius: "0.5rem",
      border: "1px solid #333",
      objectFit: "cover" as const,
      transition: "opacity 0.35s ease, transform 0.35s ease",
      opacity: status === "playing" ? 1 : 0.85,
    }),
    [isCompact, status]
  );

  const statusIndicatorStyle = useMemo(
    () => ({
      display: "inline-flex",
      alignItems: "center",
      gap: "0.6rem",
      fontWeight: 600,
      color: statusColors[status],
      transition: "opacity 0.3s ease, transform 0.3s ease",
      opacity: statusVisible ? 1 : 0,
      transform: statusVisible ? "translateY(0px)" : "translateY(-6px)",
    }),
    [status, statusVisible]
  );

  const statusDotStyle = useMemo(
    () => ({
      width: "0.75rem",
      height: "0.75rem",
      borderRadius: "999px",
      backgroundColor: statusColors[status],
      boxShadow: `0 0 0.4rem ${statusColors[status]}55`,
    }),
    [status]
  );

  const startDisabled = status === "connecting" || status === "playing";
  const stopDisabled = status === "idle";
  const submitLabel = useMemo(() => {
    if (isSubmitting) return "Starting…";
    if (status === "connecting") return "Connecting…";
    if (status === "playing") return "Streaming";
    return "Start";
  }, [isSubmitting, status]);

  const notify = useCallback(
    (message: string, variant: NotifyVariant = "info") => {
      if (message) {
        onNotify?.(message, variant);
      }
    },
    [onNotify]
  );

  const clearVideoSource = useCallback(() => {
    const node = videoRef.current;
    if (!node) return;
    node.pause();
    node.removeAttribute("src");
    node.load();
  }, []);

  const handleStop = useCallback(() => {
    if (stopDisabled) {
      return;
    }
    clearVideoSource();
    dispatch(stopStream());
  }, [clearVideoSource, dispatch, stopDisabled]);

  useEffect(
    () => () => {
      clearVideoSource();
      dispatch(stopStream());
    },
    [clearVideoSource, dispatch]
  );

  useEffect(() => {
    setStatusVisible(false);
    const timeout = window.setTimeout(() => setStatusVisible(true), 40);
    return () => window.clearTimeout(timeout);
  }, [status]);

  useEffect(() => {
    if (typeof window === "undefined" || hasHydratedConfig.current) {
      return;
    }
    try {
      const storedRaw = window.localStorage.getItem(STORAGE_KEY);
      if (!storedRaw) {
        hasHydratedConfig.current = true;
        return;
      }
      const parsed = JSON.parse(storedRaw) as Partial<NightMotionFormValues>;
      const safeValues: NightMotionFormValues = {
        username: typeof parsed.username === "string" ? parsed.username : "",
        password: typeof parsed.password === "string" ? parsed.password : "",
        streamUrl: typeof parsed.streamUrl === "string" ? parsed.streamUrl : "",
      };
      dispatch(setField({ field: "username", value: safeValues.username }));
      dispatch(setField({ field: "password", value: safeValues.password }));
      dispatch(setField({ field: "streamUrl", value: safeValues.streamUrl }));
      reset(safeValues);
    } catch {
      // Ignore malformed payloads — users can re-enter credentials.
      window.localStorage.removeItem(STORAGE_KEY);
    } finally {
      hasHydratedConfig.current = true;
    }
  }, [dispatch, reset]);

  useEffect(() => {
    const node = videoRef.current;
    if (!node) return undefined;
    const handleVideoError = () => {
      dispatch(setStatus("error"));
      dispatch(setError("Stream nelze navázat"));
      notify("Stream nelze navázat", "error");
    };
    node.addEventListener("error", handleVideoError);
    return () => {
      node.removeEventListener("error", handleVideoError);
    };
  }, [dispatch, notify]);

  const registerTextField = useCallback(
    (field: keyof NightMotionFormValues) =>
      register(field, {
        onChange: (event) => {
          dispatch(setField({ field, value: event.target.value }));
          if (status === "error") {
            dispatch(setError(null));
            dispatch(setStatus("idle"));
          }
        },
      }),
    [dispatch, register, status]
  );

  const onSubmit = handleSubmit(async (values) => {
    const payload: NightMotionFormValues = {
      username: values.username.trim(),
      password: values.password,
      streamUrl: values.streamUrl.trim(),
    };

    if (!payload.username || !payload.password || !payload.streamUrl) {
      dispatch(setError("Vyplňte všechna pole"));
      dispatch(setStatus("error"));
      notify("Vyplňte všechna pole", "error");
      return;
    }

    try {
      // Validate URL format eagerly to surface issues before attempting playback.
      // eslint-disable-next-line no-new
      new URL(payload.streamUrl);
    } catch {
      dispatch(setError("Neplatná URL adresa"));
      dispatch(setStatus("error"));
      notify("Stream nelze navázat", "error");
      return;
    }

    dispatch(setField({ field: "username", value: payload.username }));
    dispatch(setField({ field: "password", value: payload.password }));
    dispatch(setField({ field: "streamUrl", value: payload.streamUrl }));
    dispatch(setError(null));

    if (typeof window !== "undefined") {
      window.localStorage.setItem(STORAGE_KEY, JSON.stringify(payload));
    }
    notify("Nastavení uloženo", "success");

    const streamSrc = getStreamProxyUrl(payload.streamUrl, payload.username, payload.password);
    if (videoRef.current) {
      videoRef.current.src = streamSrc;
      videoRef.current.load();
    }

    try {
      await dispatch(startStream());
    } catch {
      dispatch(setStatus("error"));
      dispatch(setError("Stream nelze navázat"));
      notify("Stream nelze navázat", "error");
    }
  });

  return (
    <div style={layoutStyle}>
      <div style={{ flex: "1 1 22rem" }}>
        <FormWrapper
          title="NightMotion"
          onSubmit={onSubmit}
          isSubmitting={isSubmitting}
          isSubmitDisabled={startDisabled}
          submitLabel={submitLabel}
          onCancel={handleStop}
          cancelLabel="Stop"
          submitButtonProps={{ "data-testid": "nightmotion-start" }}
          cancelButtonProps={{ disabled: stopDisabled, "data-testid": "nightmotion-stop" }}
        >
          <div style={{ display: "flex", flexDirection: "column", gap: "0.9rem" }}>
            <label style={formFieldStyles}>
              <span>Uživatelské jméno</span>
              <input
                type="text"
                placeholder="username"
                {...registerTextField("username")}
                style={{ ...styles.input, width: "100%" }}
                autoComplete="username"
                data-testid="nightmotion-username"
              />
            </label>
            <label style={formFieldStyles}>
              <span>Heslo</span>
              <div style={{ display: "flex", gap: "0.5rem", alignItems: "center" }}>
                <input
                  type={showPassword ? "text" : "password"}
                  placeholder="password"
                  {...registerTextField("password")}
                  style={{ ...styles.input, width: "100%" }}
                  autoComplete="current-password"
                  data-testid="nightmotion-password"
                />
                <button
                  type="button"
                  onClick={() => setShowPassword((prev) => !prev)}
                  style={{
                    ...passwordToggleStyle,
                    backgroundColor: showPassword ? "#3a7bd5" : "#444",
                  }}
                  aria-label={showPassword ? "Hide password" : "Show password"}
                >
                  {showPassword ? "Hide" : "Show"}
                </button>
              </div>
            </label>
            <label style={formFieldStyles}>
              <span>Stream URL</span>
              <input
                type="url"
                placeholder="rtsp://"
                {...registerTextField("streamUrl")}
                style={{ ...styles.input, width: "100%" }}
                autoComplete="off"
                data-testid="nightmotion-stream"
              />
            </label>
          </div>
        </FormWrapper>
      </div>

      <div style={videoWrapperStyle}>
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            gap: "0.5rem",
          }}
        >
          <div style={statusIndicatorStyle} aria-live="polite" data-testid="nightmotion-status">
            <span style={statusDotStyle} />
            <span>{statusLabels[status]}</span>
          </div>
          {error && (
            <span style={{ color: statusColors.error, fontSize: "0.85rem" }} data-testid="nightmotion-error">
              {error}
            </span>
          )}
        </div>

        <video
          ref={videoRef}
          controls
          style={videoStyle}
          data-testid="nightmotion-video"
        >
          <track kind="captions" />
        </video>

        <div
          style={{
            fontSize: "0.85rem",
            color: "#b0b0b8",
            lineHeight: 1.5,
          }}
        >
          <p style={{ margin: 0 }}>
            Stream se spustí s krátkým zpožděním, aby bylo možné navázat šifrované spojení.
            Pokud se přehrávání nespustí, zkontrolujte přístupové údaje nebo adresu streamu.
          </p>
        </div>
      </div>
    </div>
  );
}
