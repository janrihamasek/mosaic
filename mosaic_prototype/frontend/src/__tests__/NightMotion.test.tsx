import React, { type ReactElement } from "react";
import { act, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";

import NightMotion from "../components/NightMotion";
import nightMotionReducer from "../store/nightMotionSlice";
import { getStreamProxyUrl } from "../api";

const notifyMock = jest.fn();

function createStore() {
  return configureStore({
    reducer: {
      nightMotion: nightMotionReducer,
    },
  });
}

function renderNightMotion() {
  const store = createStore();
  const ui: ReactElement = React.createElement(
    Provider,
    { store },
    React.createElement(NightMotion, { onNotify: notifyMock })
  );
  const utils = render(ui);
  return { store, ...utils };
}

describe("NightMotion component", () => {
  beforeAll(() => {
    Object.defineProperty(window.HTMLMediaElement.prototype, "load", {
      configurable: true,
      value: jest.fn(),
    });
    Object.defineProperty(window.HTMLMediaElement.prototype, "play", {
      configurable: true,
      value: jest.fn().mockResolvedValue(undefined),
    });
    Object.defineProperty(window.HTMLMediaElement.prototype, "pause", {
      configurable: true,
      value: jest.fn(),
    });
  });

  beforeEach(() => {
    notifyMock.mockReset();
    window.localStorage.clear();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it("renders credentials form and start button", () => {
    jest.useFakeTimers();
    renderNightMotion();
    act(() => {
      jest.advanceTimersByTime(60);
    });

    expect(screen.getByTestId("nightmotion-username")).toBeInTheDocument();
    expect(screen.getByTestId("nightmotion-password")).toBeInTheDocument();
    expect(screen.getByTestId("nightmotion-stream")).toBeInTheDocument();
    expect(screen.getByTestId("nightmotion-start")).toBeInTheDocument();
  });

  it("transitions from connecting to playing on start", async () => {
    jest.useFakeTimers();
    const { store } = renderNightMotion();
    act(() => {
      jest.advanceTimersByTime(60);
    });

    const username = "operator";
    const password = "secret";
    const streamUrl = "rtsp://camera/live";

    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });

    await user.type(screen.getByTestId("nightmotion-username"), username);
    await user.type(screen.getByTestId("nightmotion-password"), password);
    await user.type(screen.getByTestId("nightmotion-stream"), streamUrl);

    const startButton = screen.getByTestId("nightmotion-start");
    await user.click(startButton);

    await waitFor(() => expect(notifyMock).toHaveBeenCalledWith("Nastavení uloženo", "success"));
    expect(screen.getByTestId("nightmotion-status")).toHaveTextContent(/connecting/i);
    expect(store.getState().nightMotion.status).toBe("connecting");

    await act(async () => {
      jest.advanceTimersByTime(2050);
    });

    await act(async () => Promise.resolve());

    expect(store.getState().nightMotion.status).toBe("playing");
    await waitFor(() => expect(store.getState().nightMotion.status).toBe("playing"));
    await waitFor(() => expect(screen.getByTestId("nightmotion-status")).toHaveTextContent(/live/i));

    const expectedSrc = getStreamProxyUrl(streamUrl, username, password);
    const videoElement = screen.getByTestId("nightmotion-video") as HTMLVideoElement;
    expect(videoElement.src).toBe(expectedSrc);
  });

  it("stops the stream and returns to idle", async () => {
    jest.useFakeTimers();
    const { store } = renderNightMotion();
    act(() => {
      jest.advanceTimersByTime(60);
    });

    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });

    await user.type(screen.getByTestId("nightmotion-username"), "operator");
    await user.type(screen.getByTestId("nightmotion-password"), "secret");
    await user.type(screen.getByTestId("nightmotion-stream"), "rtsp://camera/live");

    await user.click(screen.getByTestId("nightmotion-start"));

    await act(async () => {
      jest.advanceTimersByTime(2050);
    });
    await act(async () => Promise.resolve());
    await waitFor(() => expect(store.getState().nightMotion.status).toBe("playing"));
    await waitFor(() => expect(screen.getByTestId("nightmotion-status")).toHaveTextContent(/live/i));

    const stopButton = screen.getByTestId("nightmotion-stop");
    expect(stopButton).not.toBeDisabled();
    await user.click(stopButton);

    await act(async () => {
      jest.advanceTimersByTime(60);
    });
    await act(async () => Promise.resolve());

    await waitFor(() => expect(store.getState().nightMotion.status).toBe("idle"));
    await waitFor(() => expect(screen.getByTestId("nightmotion-status")).toHaveTextContent(/idle/i));
    const videoElement = screen.getByTestId("nightmotion-video") as HTMLVideoElement;
    expect(videoElement.getAttribute("src")).toBeNull();
  });

  it("matches snapshot in idle state", () => {
    jest.useFakeTimers();
    const { asFragment } = renderNightMotion();

    act(() => {
      jest.advanceTimersByTime(60);
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot in playing state", async () => {
    jest.useFakeTimers();
    const { asFragment, store } = renderNightMotion();

    act(() => {
      jest.advanceTimersByTime(60);
    });

    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });

    await user.type(screen.getByTestId("nightmotion-username"), "operator");
    await user.type(screen.getByTestId("nightmotion-password"), "secret");
    await user.type(screen.getByTestId("nightmotion-stream"), "rtsp://camera/live");
    await user.click(screen.getByTestId("nightmotion-start"));

    await act(async () => {
      jest.advanceTimersByTime(2050);
    });
    await act(async () => Promise.resolve());

    await waitFor(() => expect(store.getState().nightMotion.status).toBe("playing"));
    await waitFor(() => expect(screen.getByTestId("nightmotion-status")).toHaveTextContent(/live/i));

    expect(asFragment()).toMatchSnapshot();
  });
});
