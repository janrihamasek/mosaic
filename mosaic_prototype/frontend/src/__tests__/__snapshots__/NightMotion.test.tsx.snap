// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NightMotion component matches snapshot in idle state 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: row; align-items: stretch; gap: 1.5rem; width: 100%;"
  >
    <div
      style="flex: 1 1 22rem;"
    >
      <form
        aria-busy="false"
        style="border: 1px solid #333; background-color: rgb(42, 43, 47); padding: 1.25rem; border-radius: 0.5rem; margin: 0px; box-shadow: 0 0 0.65rem rgba(0,0,0,0.45); display: flex; flex-direction: column; gap: 0.75rem;"
      >
        <header>
          <h2
            style="margin: 0px;"
          >
            NightMotion
          </h2>
        </header>
        <div
          style="display: flex; flex-direction: column; gap: 0.75rem;"
        >
          <div
            style="display: flex; flex-direction: column; gap: 0.9rem;"
          >
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Uživatelské jméno
              </span>
              <input
                autocomplete="username"
                data-testid="nightmotion-username"
                name="username"
                placeholder="username"
                style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                type="text"
              />
            </label>
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Heslo
              </span>
              <div
                style="display: flex; gap: 0.5rem; align-items: center;"
              >
                <input
                  autocomplete="current-password"
                  data-testid="nightmotion-password"
                  name="password"
                  placeholder="password"
                  style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                  type="password"
                />
                <button
                  aria-label="Show password"
                  style="padding: 0.4rem 0.6rem; background-color: rgb(68, 68, 68); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: pointer; transition: background 0.3s ease; font-size: 0.8rem; line-height: 1.4;"
                  type="button"
                >
                  Show
                </button>
              </div>
            </label>
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Stream URL
              </span>
              <input
                autocomplete="off"
                data-testid="nightmotion-stream"
                name="streamUrl"
                placeholder="rtsp://"
                style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                type="url"
              />
            </label>
          </div>
        </div>
        <div
          style="display: flex; flex-direction: row; justify-content: flex-end; align-items: center; gap: 0.75rem; margin-top: 1rem;"
        >
          <button
            data-testid="nightmotion-stop"
            disabled=""
            style="padding: 0.5rem 0.9rem; background-color: rgb(58, 59, 63); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: not-allowed; transition: background 0.3s ease; font-size: 0.95rem; line-height: 1.4; min-width: 9rem; opacity: 0.7;"
            type="button"
          >
            Stop
          </button>
          <button
            data-testid="nightmotion-start"
            style="padding: 0.5rem 0.9rem; background-color: rgb(58, 123, 213); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: pointer; transition: background 0.3s ease; font-size: 0.95rem; line-height: 1.4; min-width: 9rem; opacity: 1;"
            type="submit"
          >
            Start
          </button>
        </div>
      </form>
    </div>
    <div
      style="border: 1px solid #333; background-color: rgb(42, 43, 47); padding: 1.25rem; border-radius: 0.5rem; margin: 0.75rem 0px; box-shadow: 0 0 0.65rem rgba(0,0,0,0.45); display: flex; flex-direction: column; gap: 1rem; min-height: 20rem; flex: 1 1 24rem;"
    >
      <div
        style="display: flex; flex-direction: column; gap: 0.5rem;"
      >
        <div
          aria-live="polite"
          data-testid="nightmotion-status"
          style="display: inline-flex; align-items: center; gap: 0.6rem; font-weight: 600; color: rgb(136, 136, 136); transition: opacity 0.3s ease, transform 0.3s ease; opacity: 1; transform: translateY(0px);"
        >
          <span
            style="width: 0.75rem; height: 0.75rem; border-radius: 999px; background-color: rgb(136, 136, 136); box-shadow: 0 0 0.4rem #88888855;"
          />
          <span>
            Idle
          </span>
        </div>
      </div>
      <video
        controls=""
        data-testid="nightmotion-video"
        style="width: 100%; height: 18rem; background-color: rgb(0, 0, 0); border-radius: 0.5rem; border: 1px solid #333; object-fit: cover; transition: opacity 0.35s ease, transform 0.35s ease; opacity: 0.85;"
      >
        <track
          kind="captions"
        />
      </video>
      <div
        style="font-size: 0.85rem; color: rgb(176, 176, 184); line-height: 1.5;"
      >
        <p
          style="margin: 0px;"
        >
          Stream se spustí s krátkým zpožděním, aby bylo možné navázat šifrované spojení. Pokud se přehrávání nespustí, zkontrolujte přístupové údaje nebo adresu streamu.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NightMotion component matches snapshot in playing state 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: row; align-items: stretch; gap: 1.5rem; width: 100%;"
  >
    <div
      style="flex: 1 1 22rem;"
    >
      <form
        aria-busy="false"
        style="border: 1px solid #333; background-color: rgb(42, 43, 47); padding: 1.25rem; border-radius: 0.5rem; margin: 0px; box-shadow: 0 0 0.65rem rgba(0,0,0,0.45); display: flex; flex-direction: column; gap: 0.75rem;"
      >
        <header>
          <h2
            style="margin: 0px;"
          >
            NightMotion
          </h2>
        </header>
        <div
          style="display: flex; flex-direction: column; gap: 0.75rem;"
        >
          <div
            style="display: flex; flex-direction: column; gap: 0.9rem;"
          >
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Uživatelské jméno
              </span>
              <input
                autocomplete="username"
                data-testid="nightmotion-username"
                name="username"
                placeholder="username"
                style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                type="text"
              />
            </label>
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Heslo
              </span>
              <div
                style="display: flex; gap: 0.5rem; align-items: center;"
              >
                <input
                  autocomplete="current-password"
                  data-testid="nightmotion-password"
                  name="password"
                  placeholder="password"
                  style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                  type="password"
                />
                <button
                  aria-label="Show password"
                  style="padding: 0.4rem 0.6rem; background-color: rgb(68, 68, 68); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: pointer; transition: background 0.3s ease; font-size: 0.8rem; line-height: 1.4;"
                  type="button"
                >
                  Show
                </button>
              </div>
            </label>
            <label
              style="display: flex; flex-direction: column; gap: 0.35rem;"
            >
              <span>
                Stream URL
              </span>
              <input
                autocomplete="off"
                data-testid="nightmotion-stream"
                name="streamUrl"
                placeholder="rtsp://"
                style="padding: 0.45rem 0.75rem; margin-right: 0.5rem; border-radius: 0.25rem; border: 1px solid #555; background-color: rgb(42, 43, 47); color: rgb(230, 230, 230); width: 100%; max-width: 22rem;"
                type="url"
              />
            </label>
          </div>
        </div>
        <div
          style="display: flex; flex-direction: row; justify-content: flex-end; align-items: center; gap: 0.75rem; margin-top: 1rem;"
        >
          <button
            data-testid="nightmotion-stop"
            style="padding: 0.5rem 0.9rem; background-color: rgb(58, 59, 63); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: pointer; transition: background 0.3s ease; font-size: 0.95rem; line-height: 1.4; min-width: 9rem; opacity: 1;"
            type="button"
          >
            Stop
          </button>
          <button
            data-testid="nightmotion-start"
            disabled=""
            style="padding: 0.5rem 0.9rem; background-color: rgb(58, 123, 213); color: rgb(255, 255, 255); border-radius: 0.25rem; cursor: not-allowed; transition: background 0.3s ease; font-size: 0.95rem; line-height: 1.4; min-width: 9rem; opacity: 1;"
            type="submit"
          >
            Streaming
          </button>
        </div>
      </form>
    </div>
    <div
      style="border: 1px solid #333; background-color: rgb(42, 43, 47); padding: 1.25rem; border-radius: 0.5rem; margin: 0.75rem 0px; box-shadow: 0 0 0.65rem rgba(0,0,0,0.45); display: flex; flex-direction: column; gap: 1rem; min-height: 20rem; flex: 1 1 24rem;"
    >
      <div
        style="display: flex; flex-direction: column; gap: 0.5rem;"
      >
        <div
          aria-live="polite"
          data-testid="nightmotion-status"
          style="display: inline-flex; align-items: center; gap: 0.6rem; font-weight: 600; color: rgb(46, 204, 113); transition: opacity 0.3s ease, transform 0.3s ease; opacity: 0; transform: translateY(-6px);"
        >
          <span
            style="width: 0.75rem; height: 0.75rem; border-radius: 999px; background-color: rgb(46, 204, 113); box-shadow: 0 0 0.4rem #2ecc7155;"
          />
          <span>
            Live
          </span>
        </div>
      </div>
      <video
        controls=""
        data-testid="nightmotion-video"
        src="http://127.0.0.1:5000/api/stream-proxy?url=rtsp%3A%2F%2Fcamera%2Flive&username=operator&password=secret"
        style="width: 100%; height: 18rem; background-color: rgb(0, 0, 0); border-radius: 0.5rem; border: 1px solid #333; object-fit: cover; transition: opacity 0.35s ease, transform 0.35s ease; opacity: 1;"
      >
        <track
          kind="captions"
        />
      </video>
      <div
        style="font-size: 0.85rem; color: rgb(176, 176, 184); line-height: 1.5;"
      >
        <p
          style="margin: 0px;"
        >
          Stream se spustí s krátkým zpožděním, aby bylo možné navázat šifrované spojení. Pokud se přehrávání nespustí, zkontrolujte přístupové údaje nebo adresu streamu.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
